internal/providers/terraform/aws/rds_cluster.go

```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
+				costComponents = append(costComponents, auroraBackupStorageCostComponent(region, &totalBackupStorageGB, databaseEngine))
+			}
+		} else {
+			var unknown *decimal.Decimal
 
-		totalBackupStorageGB := calculateBackupStorage(snapShotStorageSizeGB, backupStorageRetention)
-		if totalBackupStorageGB.GreaterThan(decimal.Zero) {
-			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, totalBackupStorageGB, databaseEngine))
+			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, unknown, databaseEngine))
 		}
 	}
 
 	if databaseEngineMode != "Serverless" && !strings.Contains(d.Get("engine").String(), "postgresql") {
-		averageStatements := decimal.Zero
-		if u != nil && u.Get("average_statements_per_hr").Exists() {
-			averageStatements = decimal.NewFromInt(u.Get("average_statements_per_hr").Int())
-		}
+		var averageStatements, backtrackChangeRecords, backtrackWindowHours *decimal.Decimal
+		if u != nil && averageStatementsPerHrExists(u) && changeRecordsPerStatementExists(u) && backtrackWindowHrsExists(u) {
+			averageStatements = decimalPtr(decimal.NewFromInt(u.Get("average_statements_per_hr").Int()))
+			backtrackChangeRecords = decimalPtr(decimal.NewFromFloat(u.Get("change_records_per_statement").Float()))
+			backtrackWindowHours = decimalPtr(decimal.NewFromInt(u.Get("backtrack_window_hrs").Int()))
 
-		backtrackChangeRecords := decimal.Zero
-		if u != nil && u.Get("change_records_per_statement").Exists() {
-			backtrackChangeRecords = decimal.NewFromFloat(u.Get("change_records_per_statement").Float())
-		}
+			totalBacktrackChangeRecords := calculateBacktrack(*averageStatements, *backtrackChangeRecords, *backtrackWindowHours)
 
-		backtrackWindowHours := decimal.Zero
-		if u != nil && u.Get("backtrack_window_hrs").Exists() {
-			backtrackWindowHours = decimal.NewFromInt(u.Get("backtrack_window_hrs").Int())
-		}
+			if totalBacktrackChangeRecords.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBacktrackCostComponent(region, &totalBacktrackChangeRecords))
+			}
+		} else {
+			var unknown *decimal.Decimal
 
-		totalBacktrackChangeRecords := calculateBacktrack(averageStatements, backtrackChangeRecords, backtrackWindowHours)
-		if totalBacktrackChangeRecords.GreaterThan(decimal.Zero) {
-			costComponents = append(costComponents, auroraBacktrackCostComponent(region, totalBacktrackChangeRecords))
+			costComponents = append(costComponents, auroraBacktrackCostComponent(region, unknown))
 		}
 	}
 
-	snapshotExportSizeGB := decimal.Zero
+	var snapshotExportSizeGB *decimal.Decimal
 	if u != nil && u.Get("snapshot_export_size_gb").Exists() {
-		snapshotExportSizeGB = decimal.NewFromFloat(u.Get("snapshot_export_size_gb").Float())
-	}
+		snapshotExportSizeGB = decimalPtr(decimal.NewFromFloat(u.Get("snapshot_export_size_gb").Float()))
+
+		if snapshotExportSizeGB.GreaterThan(decimal.Zero) {
+			costComponents = append(costComponents, &schema.CostComponent{
```

Comment:
We can use only one `costComponents = append...` here, just take out this part of code outside of `if`, then `else` block not needed